src/components/Layout/Header/NavBar/UserMenu.tsx

```diff
@@ -89,14 +91,38 @@ type WalletButtonProps = {
   onConnect: () => void
 } & Pick<InitialState, 'walletInfo'>
 
-const WalletButton = ({ isConnected, walletInfo, onConnect }: WalletButtonProps) => {
+const WalletButton: FC<WalletButtonProps> = ({ isConnected, walletInfo, onConnect }) => {
+  const [walletLabel, setWalletLabel] = useState('')
+  const bgColor = useColorModeValue('gray.300', 'gray.800')
+
+  useEffect(() => {
+    setWalletLabel('')
+    if (!walletInfo || !walletInfo.meta) return setWalletLabel('')
+    if (walletInfo.meta.address) return setWalletLabel(walletInfo.meta.address)
+    if (walletInfo.meta.label) return setWalletLabel(walletInfo.meta.label)
+  }, [walletInfo])
+
   return Boolean(walletInfo?.deviceId) ? (
     <Button
       onClick={onConnect}
       leftIcon={<WalletImage walletInfo={walletInfo} />}
       rightIcon={isConnected ? undefined : <WarningTwoIcon ml={2} w={3} h={3} color='yellow.500' />}
     >
-      {walletInfo?.name}
+      {walletLabel !== '' ? (
```

Comment:
Nitpick: Since `walletLabel` can only be a string and an empty string if falsy: `walletLabel ? left : right`

Reply:
fixed